### DISCUSSION TITLE
Loading pre-trained data into slicer

### DISCUSSION THREAD

**Discussion by D123Roy** [Project-MONAI/tutorials#1009] [General]
I am working with a pre-trained model (trained on a custom dataset) for interactive segmentation. I am using 3D Slicer for the interactive segment purpose.- 
I developed a pre-trained weights file (.pth) by training with the following link - https://colab.research.google.com/github/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb#scrollTo=Wv2UxnI2gc-K
On adding the path of the file to deepedit.py file in Slicer apps -> lib -> config folder, and then running the model the error is -

Run time error : 
Given groups=1,weight of size[32,5,3,3,3], expected input[1,8,128,128,128] to have 5 channels, but got 8 channels instead.

I am clueless. Kindly help

**Comment by KumoLiu** [comment]
@D123Roy, BRATS dataset has 5 classes (including background), the error message indicated that your custom dataset has 8 channels which caused the runtime error. If you want to use this pretrained model for your 8-channel dataset, you need to modify the channel to 8. You can refer to [this link ](https://pytorch.org/tutorials/beginner/transfer_learning_tutorial.html) for more details on transfer learning.

**Reply by D123Roy** [reply]
@KumoLiu Thanks for your reply. Actually I am testing the model with BRATS dataset itself. But the error persists. I changed the labels as per the BRATS models (leading to 4 channels)

My doubt is - Is it sufficient to just add the path of my pre-trained model (.pth) to the loaded conf model in Slicer
Do I need to make more changes ? I am stuck with this for a long time

**Reply by D123Roy** [reply]
ie the images are DecathlonDataset with size as [4, 240, 240, 155]

**Reply by KumoLiu** [reply]
Hi @D123Roy, I'm not very familiar with how to import models in 3d slicer, but from the error message it looks like there is still something wrong with your data dimensions, you could check your data first.
And for how to load the model in Slicer, can @tangy5 help give some suggestions, thanks!

**Comment by tangy5** [comment]
Hi @D123Roy , looking at the logs, it seems the way for loading the input data has some errors. No sure, but I guess the way of concatenation might be wrong in your case, since the input data should be 4 channels per data, showing 8 probably means you are concatenating two images in the second channel instead of the batch channel.  The tutorial should has 4 channels as input, not sure why 5 is required in your case, you could try double check this code section when defining network:

```
model = SegResNet(
    blocks_down=[1, 2, 2, 4],
    blocks_up=[1, 1, 1],
    init_filters=16,
    in_channels=4,
    out_channels=3,
    dropout_prob=0.2,
).to(device)
```

In addition, for loading your own pretrained model, there should be no problem based on your description, you could see there are model downloading process when you are starting MONAI label server, to double confirm, you can check whether your code can load the different path/filename or you can just replace the downloaded model with your own. 

Hope these helps. 

BTW, it's better to discuss the Slicer and MONAI Label issues in the MONAI Label repo here: 
https://github.com/Project-MONAI/MONAILabel/discussions
Welcome discuss and thanks for using MONAI and MONAI Label.
